File: challenges/day1.js
const fs = require('fs');
const moduleMasses = fs.readFileSync('./inputs/day1.txt').toString().split('\n');

const answer = moduleMasses.reduce((acc, n) => acc + fuelNeeded(n), 0);
console.log(answer);

function fuelNeeded(n) {
  const initialFuelCalc = Math.floor(n / 3) - 2;
  return initialFuelCalc <= 0 ? 0 : initialFuelCalc + fuelNeeded(initialFuelCalc);
}

File: challenges/day1.py
import math as math
with open('./inputs/day1.txt', 'r') as f:
  moduleMasses = f.read().splitlines()

def fuelCalculation(mass):
  fuel = math.floor(int(mass) / 3) - 2
  if fuel <= 0: return 0
  return fuel + fuelCalculation(fuel)

sum = 0
for m in moduleMasses: 
  sum += fuelCalculation(m)
print(sum)
